apollo-gradle-plugin-incubating/src/main/kotlin/com/apollographql/apollo/gradle/internal/ApolloPlugin.kt

```diff
@@ -260,11 +262,35 @@ open class ApolloPlugin : Plugin<Project> {
       registerCodegenTasks(project, apolloExtension)
 
       registerDownloadSchemaTasks(project, apolloExtension)
+
+      checkVersions(project)
+    }
+
+    data class Dep(val name: String, val version: String?)
+
+    fun getDeps(configurations: ConfigurationContainer): List<Dep> {
```

Comment:
Does this mean that we resolve dependencies during configuration time? If so, I wonder if that can be done lazily.

Reply:
I was under the impression that gradle would resolve the dependencies during configuration in all cases. If not then it's going to add some processing to each and every gradle build :/.

What do you mean by `lazily`? maybe just run the check along the `compileApolloSources` tasks? And not if it's already up-to-date ?

Reply:
I'm not sure about the solution itself myself. Maybe do this when one of our tasks started.

Reply:
On my machine, once the gradle cache warmed up, it takes 7ms to execute this check, maybe it's not too bad ?

Reply:
I don't think this is a blocker anyways. I think this becomes little annoyance if you have really bad connection or start the project offline etc.